### ISSUE TITLE
fedora36: srw NSLS-II ESM Beamline example simulations on the beamline tab don't work

### ISSUE THREAD

**Issue by e-carlin** [radiasoft/sirepo#5276]
Coherent results works. Partially coherent and 3D Beamline don't

**Comment by e-carlin** [comment]
```
multiElectronAnimation (master)$ python -X faulthandler parameters.py
Fatal Python error: Segmentation fault

Current thread 0x00007f33ca1eb000 (most recent call first):
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.10/site-packages/srwlib.py", line 9015 in srwl_wfr_emit_prop_multi_e
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.10/site-packages/srwl_bl.py", line 1933 in calc_wfr_emit_prop_me
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.10/site-packages/srwl_bl.py", line 3226 in calc_all
  File "/home/vagrant/src/radiasoft/sirepo/run/user/VMLVAwKb/srw/nFiYOlsa/multiElectronAnimation/parameters.py", line 578 in main
  File "/home/vagrant/src/radiasoft/sirepo/run/user/VMLVAwKb/srw/nFiYOlsa/multiElectronAnimation/parameters.py", line 580 in <module>
```
[parameters.py.txt](https://github.com/radiasoft/sirepo/files/10355785/parameters.py.txt)

**Comment by robnagler** [comment]
You need to point gdb at the source. Then you need a `bt` for the error. We can debug together.

**Comment by e-carlin** [comment]
Some notes:
I compiled srw against python 3.7, 3.8, 3.9, and 3.10. All worked except 3.10. I also tried the commit of srw that we last built from vs the most recent. That didn't make a difference. Both worked on 3.9 and segfaulted on 3.10.

There are breaking changes related to the c-api in python 3.10 https://docs.python.org/3/whatsnew/3.10.html . 

Here are avenues I've explored so far that didn't work:
- PyUnicode removal: The functions srw uses are still good
- Limited C-API with debug build: We set  --with-pydebug. Removing it made no difference. I don't know for sure that the build happened without something else setting --with-pydebuge so this could still be the issues but seems unlikely.
- Tried adding `#define PY_SSIZE_T_CLEAN` before Python.h includes. This is recommended by the [python docs](https://docs.python.org/3/c-api/intro.html). I forgot that I had to [do this for radia](https://github.com/radiasoft/download/commit/c7746b491f20116265fdd1e02366853f0b4b95de) to work on py3.10. This feels like a promising lead. The [porting to python 3.10](https://docs.python.org/3/whatsnew/3.10.html#id2) docs mention it, radia is another ochubar code, and I believe the problem we are running into is in parsing arguments from python which this has something to do with. Doing the same fix as I did in radia didn't work but I still think this is worth exploring more.
